pkg/operators/oci-handler/oci.go

```diff
@@ -266,11 +266,20 @@ func (o *OciHandlerInstance) init(gadgetCtx operators.GadgetContext) error {
 }
 
 func (o *OciHandlerInstance) Start(gadgetCtx operators.GadgetContext) error {
+	started := []operators.DataOperatorInstance{}
+
 	for _, opInst := range o.imageOperatorInstances {
 		err := opInst.Start(o.gadgetCtx)
 		if err != nil {
-			o.gadgetCtx.Logger().Errorf("starting operator %q: %v", opInst.Name(), err)
+			// Stop all started operators
+			for _, startedOp := range started {
+				startedOp.Stop(o.gadgetCtx)
```

Comment:
Actually, why not to call `o.Stop()`. The `ImageOperatorInstance.Stop` implementations should be robust enough to do nothing if they haven't been initialized, don't you think?

Reply:
I know we have been doing this for the gadgets and so on but IMO there is not any reason why Stop() should be called on something that hasn't been started. It doesn't make sense and just complicates the implementation of those operators.

Reply:
Calling `Stop` complicates the logic of operators (They have to check nil before calling Close, Stop, etc ... Already done by the ebpf operator and most operators). Stopping only the started operators complicate the logic of the caller (You had the created a temporary slice to know which operators were started). I think it's a matter of decide where we want to put the "complexity" 😅. I'm ok either way, we just need to be consistent and use the same approach everywhere from now on.

Reply:
It's better to complicate the logic in one place (oci operator) than in many (all image operator), specially given that users of the API could implement their own image operators.